src/runtime/disco/loader.cc

```diff
@@ -60,6 +61,10 @@ class ShardLoaderObj : public Object {
     const FileRecord* file;
     const ParamRecord* param;
     int shard_dim;
+    PackedFunc f_reorder_fp16_;
```

Comment:
Why do we need 3 different functions for different dtypes? Why can't we pass different packed func in shard info when dtypes are differnt?

Reply:
Can we rename to f_preprocess, because it's more general?